packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -2011,21 +2019,18 @@ class TextSelectionGestureDetectorBuilder {
       || kind == PointerDeviceKind.stylus;
 
     // Handle shift + click selection if needed.
-    final bool isShiftPressedValid = _isShiftPressed && renderEditable.selection?.baseOffset != null;
+    final bool isShiftPressed = _containsShift(details.keysPressedOnDown);
+    final bool isShiftPressedValid = isShiftPressed && renderEditable.selection?.baseOffset != null;
```

Comment:
It is a bit confusing for the name `isShiftPressedValid` can you rename it or add some comment on why the shiftpressed is only valid when there is a existing selection in renderEditable?

Reply:
Added a comment.